### PR TITLE
fix(@desktop/wallet): Display balance for all accounts

### LINKED ISSUE #12001: Balance history is not fetched on "All accounts" button clicked
# Bug Report

## Description
All_accounts_button

After clicking "All accounts" button -> click a token (Eth, for example) -> "Balance" tab -no balance history. Erro in the log 
`err="invalid argument 1: hex string has length 0, want 40 for common.Address"`

#### Expected behavior
See balance history for all accounts combined
#### Actual behavior
No balance history displayed
### Additional Information

- Status desktop version: 0.13.3 && master (43f49c0b2d815642d4f667f1dce27b80cbcfafc4)
- Operating System: Ubuntu 20.04

### PR DESCRIPTION
**Pull request by Cuteivist**
fixes #12001 

status-go: https://github.com/status-im/status-go/pull/4261

### What does the PR do

* Removed wallet service reference not needed anymore
* Handled show all addresses balance
* Added fill for all addresses filter change to be able to pass all addresses to backend

### Affected areas

Wallet / assets / balance

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/11396062/1b198548-5b23-4442-9985-175796c4b643

### CODE REVIEW

**Review by stefandunca — APPROVED** [review]
🤩 Good to see all balances working. Great job.

**Comment by glitchminer** [comment]
Attempted this with 2 watch accounts, a balance was shown for 1 account and the other was displayed as 0 in the chart while being $13.98 in the header.

image

File: src/app/modules/main/wallet_section/all_tokens/module.nim [hunk 1]
```diff
@@ -68,8 +69,9 @@ method getHistoricalDataForToken*(self: Module, symbol: string, currency: string
 method tokenHistoricalDataResolved*(self: Module, tokenDetails: string) =
   self.view.setTokenHistoricalDataReady(tokenDetails)
 
-method fetchHistoricalBalanceForTokenAsJson*(self: Module, address: string, tokenSymbol: string, currencySymbol: string, timeIntervalEnum: int) =
-  self.controller.fetchHistoricalBalanceForTokenAsJson(address, tokenSymbol, currencySymbol,timeIntervalEnum)
+method fetchHistoricalBalanceForTokenAsJson*(self: Module, address: string, allAddresses: bool, tokenSymbol: string, currencySymbol: string, timeIntervalEnum: int) =
+  let addresses = if allAddresses: self.addresses else: @[address]
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply src/app/modules/main/wallet_section/all_tokens/module.nim:73]: @IvanBelyakoff If I remember correctly, this is, by design, a shortcut of not moving all addresses through the QML layer but using a flag and interpreting the flag at the nim part of the presentation layer. Therefore we pass only the state from QML and then source the data. It looks fine to me. If we don't have to display all the addresses, why push them to the QML layer? Is this what you asked?

[the remark this answers is not in the document]
Comment (stefandunca) [reply src/app/modules/main/wallet_section/all_tokens/module.nim:74]: @IvanBelyakoff, I think I understand your point of view on having a clear intention for the controller API/calls on what addresses will be displayed.

The current approach of stating the intention at a higher level of abstraction still does its job. The QML account switcher knows it has to display all enabled addresses or one of the identities passed by the model. I think this is enough and doesn't need more responsibility for which specific addresses are included in this all-address state. This way, we move the responsibility to the controller (module) level without sharing it with the top of the presentation layer (QML).

This confusing API could have been better solved with a "WalletViewState" item to be passed to the specific controller to encapsulate the state intention better. However, the nim bridge makes it impractical at this moment.

**Pull request merged**